src/data/mv_polynomial.lean

```diff
@@ -1292,4 +1323,129 @@ end
 
 end equiv
 
+/-! lemmas about `pderivative v`, the partial derivative with respect to `v : σ`-/
+section pderivative
+
+variables {R : Type} [comm_ring R]
+variable {S : Type}
+
+/-- pderivative v p is the partial derivative of p with respect to v -/
+def pderivative (v : S) (p : mv_polynomial S R) : mv_polynomial S R :=
+p.sum (λ A B, monomial (erase v A + single v (A v - 1)) (B * (A v)))
+
+lemma pderivative_monomial {a : R} {v : S} {n : ℕ} :
+  pderivative v (C a * (X v)^n) = C (a * n) * (X v)^(n - 1) :=
+begin
+  rw [←single_eq_C_mul_X, ←single_eq_C_mul_X, pderivative, monomial, sum_single_index],
+  { rw [erase_single, zero_add, single_eq_same]},
+  rw [erase_single, zero_mul, zero_add],
+  exact monomial_zero,
+end
+
+@[simp]
+lemma pderivative_C {a : R} {v : S} : pderivative v (C a) = 0 :=
+begin
+  rw [(show C a = C a * (X v)^0, by rw [pow_zero, mul_one]),
```

Comment:
I don't understand mathlib conventions. Is `Z^0` and not `Z ^ 0` OK?

Reply:
For powers I tend to not care so much... I don't know why...